src/fingerprint_bond.cpp

```diff
@@ -0,0 +1,917 @@
+/* -*- c++ -*- ----------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   http://lammps.sandia.gov, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+/*  ----------------------------------------------------------------------
+   Contributing authors: Christopher Barrett (MSU) barrett@me.msstate.edu
+   	   	   	   	   	     Doyl Dickel (MSU) doyl@me.msstate.edu
+    ----------------------------------------------------------------------*/
+/*
+“The research described and the resulting data presented herein, unless
+otherwise noted, was funded under PE 0602784A, Project T53 "Military
+Engineering Applied Research", Task 002 under Contract No. W56HZV-17-C-0095,
+managed by the U.S. Army Combat Capabilities Development Command (CCDC) and
+the Engineer Research and Development Center (ERDC).  The work described in
+this document was conducted at CAVS, MSU.  Permission was granted by ERDC
+to publish this information. Any opinions, findings and conclusions or
+recommendations expressed in this material are those of the author(s) and
+do not necessarily reflect the views of the United States Army.​”
+
+DISTRIBUTION A. Approved for public release; distribution unlimited. OPSEC#4918
+ */
+
+#include "fingerprint_bond.h"
+
+
+using namespace LAMMPS_NS;
+
+Fingerprint_bond::Fingerprint_bond(PairRANN *pair) : Fingerprint(pair)
+{
+	n_body_type = 3;
+	dr = 0;
+	re = 0;
+	rc = 0;
+	alpha_k = new double [1];
+	alpha_k[0] = -1;
+	k = 0;
+	m = 0;
+	id = -1;
+	style = "bond";
+	atomtypes = new int [n_body_type];
+	empty = true;
+	pair->allscreen = false;
+}
+
+Fingerprint_bond::~Fingerprint_bond(){
+	delete [] alpha_k;
+	delete [] atomtypes;
+	delete [] expcuttable;
+	delete [] dfctable;
+	for (int i=0;i<(m*(m+1))>>1;i++){
+		delete [] coeff[i];
+		delete [] coeffx[i];
+		delete [] coeffy[i];
+		delete [] coeffz[i];
+		delete [] Mf[i];
+	}
+	delete [] coeff;
+	delete [] coeffx;
+	delete [] coeffy;
+	delete [] coeffz;
+	delete [] Mf;
+	delete [] rinvsqrttable;
+}
+
+bool Fingerprint_bond::parse_values(char * constant, char * line1){
+	char **words=new char *[MAXLINE];
```

Comment:
since we moved to C++11 as minimum C++ standard requirement, we are also switching away from some of the C-string string functions and use the C++ std::string class instead. Comparisons become more readable, search and substitution requires less code and functions that are not thread-safe like strtok() can be avoided. strtok() in particular is something that we want to get rid of and not add new uses to the code.
For parsing/processing text files there are utility functions in the utils:: namespace and particular there are the Tokenizer and ValueTokenizer classes that are already applied to a bunch of parsers for reading potential files.

Reply:
I've removed all uses of strtok and used the Tokenizer instead. I did not use the text file reader because it has the line limit of 1024 hard coded. This conflicts with my potential file convention which has formatted matrices. The weight matrices can have more than 60 columns for alloys which with 16 digit precision will exceed that line length.